src/coreclr/tests/src/JIT/opt/InstructionCombining/CondToBranchlessExpr.cs

```diff
@@ -0,0 +1,201 @@
+    public int Test20(int a) => a != 42 ? Cns(2) : Cns(2);
+    public int Test21(int a) => a != 42 ? Cns(3) : Cns(2);
+    public int Test22(int a) => a != 42 ? Cns(4) : Cns(2);
+    public int Test23(int a) => a != 42 ? Cns(4) : Cns(6);
+    public int Test24(int a) => a != 42 ? Cns(int.MaxValue) : Cns(int.MaxValue - 1);
+    public int Test25(int a) => a != 42 ? Cns(int.MaxValue) : Cns(int.MaxValue);
+    public int Test26(int a) => a != 42 ? Cns(int.MaxValue - 1) : Cns(int.MaxValue);
+
+    public int Test27(int a) => a > 42 ? Cns(0) : Cns(1);
+    public int Test28(int a) => a > 42 ? Cns(0) : Cns(0);
+    public int Test29(int a) => a > 42 ? Cns(1) : Cns(0);
+    public int Test30(int a) => a > 42 ? Cns(0) : Cns(-1);
+    public int Test31(int a) => a > 42 ? Cns(-1) : Cns(0);
+    public int Test32(int a) => a > 42 ? Cns(1) : Cns(2);
+    public int Test33(int a) => a > 42 ? Cns(2) : Cns(2);
+    public int Test34(int a) => a > 42 ? Cns(3) : Cns(2);
+    public int Test35(int a) => a > 42 ? Cns(4) : Cns(2);
+    public int Test36(int a) => a > 42 ? Cns(4) : Cns(6);
+    public int Test37(int a) => a > 42 ? Cns(int.MaxValue) : Cns(int.MaxValue - 1);
+    public int Test38(int a) => a > 42 ? Cns(int.MaxValue) : Cns(int.MaxValue);
+    public int Test39(int a) => a > 42 ? Cns(int.MaxValue - 1) : Cns(int.MaxValue);
+
+    // GT_TEST_/test instruction
+    public int Test40(int a) => a == 0 ? Cns(0) : Cns(1);
+    public int Test41(int a) => a == 0 ? Cns(0) : Cns(0);
+    public int Test42(int a) => a == 0 ? Cns(1) : Cns(0);
+    public int Test43(int a) => a == 0 ? Cns(0) : Cns(-1);
+    public int Test44(int a) => a == 0 ? Cns(-1) : Cns(0);
+    public int Test45(int a) => a == 0 ? Cns(1) : Cns(2);
+    public int Test46(int a) => a == 0 ? Cns(2) : Cns(2);
+    public int Test47(int a) => a == 0 ? Cns(3) : Cns(2);
+    public int Test48(int a) => a == 0 ? Cns(4) : Cns(2);
+    public int Test49(int a) => a == 0 ? Cns(4) : Cns(6);
+    public int Test50(int a) => a == 0 ? Cns(int.MaxValue) : Cns(int.MaxValue - 1);
+    public int Test51(int a) => a == 0 ? Cns(int.MaxValue) : Cns(int.MaxValue);
+    public int Test52(int a) => a == 0 ? Cns(int.MaxValue - 1) : Cns(int.MaxValue);
+
+    public int Test53(int a) => a != 0 ? Cns(0) : Cns(1);
+    public int Test54(int a) => a != 0 ? Cns(0) : Cns(0);
+    public int Test55(int a) => a != 0 ? Cns(1) : Cns(0);
+    public int Test56(int a) => a != 0 ? Cns(0) : Cns(-1);
+    public int Test57(int a) => a != 0 ? Cns(-1) : Cns(0);
+    public int Test58(int a) => a != 0 ? Cns(1) : Cns(2);
+    public int Test59(int a) => a != 0 ? Cns(2) : Cns(2);
+    public int Test60(int a) => a != 0 ? Cns(3) : Cns(2);
+    public int Test61(int a) => a != 0 ? Cns(4) : Cns(2);
+    public int Test62(int a) => a != 0 ? Cns(4) : Cns(6);
+    public int Test63(int a) => a != 0 ? Cns(int.MaxValue) : Cns(int.MaxValue - 1);
+    public int Test64(int a) => a != 0 ? Cns(int.MaxValue) : Cns(int.MaxValue);
```

Comment:
>Cns(int.MaxValue) : Cns(int.MaxValue); [](start = 41, length = 38)

Here and elsewhere I think you meant by adding `Cns(int.MaxValue) : Cns(int.MinValue)` ? If not, that will also be an interesting case to test.